src/lines_lf.rs

```diff
@@ -122,36 +133,41 @@ fn count_breaks_impl<T: ByteChunk>(text: &[u8]) -> usize {
     if text.len() < T::SIZE {
         // Bypass the more complex routine for short strings, where the
         // complexity hurts performance.
-        text.iter().map(|byte| (*byte == 0x0A) as usize).sum()
+        count_lf(text)
     } else {
         // Get `middle` so we can do more efficient chunk-based counting.
         let (start, middle, end) = unsafe { text.align_to::<T>() };
 
         let mut count = 0;
 
         // Take care of unaligned bytes at the beginning.
-        for byte in start.iter() {
-            count += (*byte == 0x0A) as usize;
+        count += count_lf(start);
+
+        // Take care of the middle bytes in big chunks. Loop unrolled.
+        for chunks in middle.chunks_exact(4) {
+            let val1 = chunks[0].cmp_eq_byte(LF);
+            let val2 = chunks[1].cmp_eq_byte(LF);
+            let val3 = chunks[2].cmp_eq_byte(LF);
+            let val4 = chunks[3].cmp_eq_byte(LF);
+            count += val1.add(val2).add(val3.add(val4)).sum_bytes();
         }
 
-        // Take care of the middle bytes in big chunks.
-        for chunks in middle.chunks(T::MAX_ACC) {
-            let mut acc = T::zero();
-            for chunk in chunks.iter() {
-                acc = acc.add(chunk.cmp_eq_byte(0x0A));
-            }
-            count += acc.sum_bytes();
+        // Chunk remainder
+        let mut acc = T::zero();
+        for chunk in middle.chunks_exact(4).remainder() {
+            acc = acc.add(chunk.cmp_eq_byte(LF));
         }
+        count += acc.sum_bytes();
 
         // Take care of unaligned bytes at the end.
-        for byte in end.iter() {
-            count += (*byte == 0x0A) as usize;
-        }
-
-        count
+        count + count_lf(end)
     }
 }
 
+fn count_lf(text: &[u8]) -> usize {
+    text.iter().filter(|&b| *b == LF).count()
```

Comment:
I don't think it adds much clarity to split this out into a separate function.  I preferred the code as it was before, with the inline for loops.

Reply:
Or maybe just replace all the places where this would be called with `blahblah.iter().map(|byte| (*byte == 0x0A) as usize).sum()`, since that's what was already used further up.